Deepgram/Clients/Speak/v1/WebSocket/WebSocketMessage.cs

```diff
@@ -4,9 +4,30 @@
 
 namespace Deepgram.Clients.Speak.v1.WebSocket;
 
-internal readonly struct WebSocketMessage(byte[] message, WebSocketMessageType type)
+internal readonly struct WebSocketMessage
 {
-    public ArraySegment<byte> Message { get; } = new ArraySegment<byte>(message);
+    public WebSocketMessage(byte[] message, WebSocketMessageType type)
+        : this(message, type, Constants.UseArrayLengthForSend)
+    {
+    }
 
-    public WebSocketMessageType MessageType { get; } = type;
+    public WebSocketMessage(byte[] message, WebSocketMessageType type, int length)
+    {
+        if (length != Constants.UseArrayLengthForSend || length < Constants.UseArrayLengthForSend)
+        {
```

Comment:
What is the purpose of the || condition?  If length is < -1, then it is a negative number, what does it mean to pass a negative number to ArraySegment?  Perhaps you meant the following instead?

if (length != Constants.UseArrayLengthForSend && length <= message.Length)  // ensure provided length is valid

Reply:
If you pass a negative number (which is invalid), it just takes the length of the buffer. We could also throw an exception.

Reply:
But a negative length will go into the top half of the if and will be passed to ArraySegment.

Reply:
Seems the following would be best:
if (length != Constants.UseArrayLengthForSend && length <= message.Length && length > 0) // if length is provided ensure it is valid

Reply:
I see your point. will make an update for more bumper rails in a follow up PR. In the meantime, I hope that someone would not put in a negative length.